File: vinzi.jsonMgt/src/main/clojure/vinzi/jsonMgt/hib/clj/Commit.clj
(ns vinzi.jsonMgt.hib.clj.Commit
    (:use vinzi.hib-connect.globals)
    (:import vinzi.jsonMgt.hib.Commit))

(defrecord Commit_clj [id track_id datetime contents])

(defn create-Commit [track_id datetime contents]
  ;; TO DO:  insert type-checking (java-objects are strictly typed)
  (Commit_clj. nil track_id datetime contents))

(defn clone-Commit [cRec]
  ;; TO DO:  insert type-checking cRec should be same type/class as target
  (create-Commit (:track_id cRec) (:datetime cRec) (:contents cRec)))

(defn Commit-to-java [cRec]
	(Commit. (:id cRec) (:track_id cRec) (:datetime cRec) (:contents cRec) ))


(add-to-java (Commit_clj. 0 1 2 3) Commit-to-java)

(defn Commit-to-clj [jRec]
	(Commit_clj. (.getId jRec) (.getTrack_id jRec) (.getDatetime jRec) (.getContents jRec) ))


(add-to-clj (Commit.) Commit-to-clj)

File: vinzi.jsonMgt/src/main/java/vinzi/jsonMgt/hib/Commit.java
package vinzi.jsonMgt.hib;

public class Commit {
    public Long id;
    public Long getId() { return id;}
    protected void setId(Long id)  {this.id = id; };


    public Long track_id;
    public Long getTrack_id() { return track_id;}
    public void setTrack_id(Long track_id)  {this.track_id = track_id; };


    public java.util.Date datetime;
    public java.util.Date getDatetime() { return datetime;}
    public void setDatetime(java.util.Date datetime)  {this.datetime = datetime; };


    public String contents;
    public String getContents() { return contents;}
    public void setContents(String contents)  {this.contents = contents; };


    public Commit() {};

    public Commit(Long id, Long track_id, java.util.Date datetime, String contents) {
	this.id = id;
	this.track_id = track_id;
	this.datetime = datetime;
	this.contents = contents;
	return;};

   public String toString() {
        return " id=" + this.id + " track_id=" + this.track_id + " datetime=" + this.datetime + " contents=" + this.contents;};

};
